stm/stm.ml

```diff
@@ -882,7 +844,6 @@ end = struct (* {{{ *)
         | Some _, None -> (e, info)
         | Some (_,at), Some id -> (e, Stateid.add info ~valid:id at) in
       if cache = `Yes || cache = `Shallow then freeze_invalid id ie;
-      Hooks.(call unreachable_state id ie);
```

Comment:
This is used by PIDE, your patch to pidetop does remove the hook call but it is needed.
It is like the `cancel` messages of serapi, PIDE needs to be notifies of each and every state (exec id) that is processed (or cancelled for processing).